server/actions_artifacts.go

```diff
@@ -187,7 +188,11 @@ func (s *RegistryServer) ListArtifacts(ctx context.Context, req *rpc.ListArtifac
 			if err != nil {
 				return nil, invalidArgumentError(err)
 			}
-			if !out.Value().(bool) {
+			v, ok := out.Value().(bool)
+			if !ok {
+				return nil, invalidArgumentError(errors.New("expression does not evaluate to a boolean"))
+			}
+			if !v {
 				continue
```

Comment:
It's noteworthy that this code is duplicated five times - I'd support factoring this out into common filtering code, but wanted to get this working first before formalizing that structure.